plugins/firehydrant/src/components/ServiceDetailsCard/ServiceDetailsCard.tsx

```diff
@@ -149,14 +150,14 @@ export const ServiceDetailsCard = () => {
   const startDate = DateTime.now().minus({ days: 30 }).toUTC();
   const endDate = DateTime.now().toUTC();
 
+  // The service name is provided by an annotation or a Backstage generated service name.
   // The Backstage service name in FireHydrant is a unique formatted string
   // that requires the entity's kind, name, and namespace.
-  const fireHydrantServiceName = `${entity?.kind}:${
-    entity?.metadata?.namespace ?? 'default'
-  }/${entity?.metadata?.name}`;
+  const fireHydrantServiceName = getFireHydrantServiceName(entity);
 
   const { loading, value, error } = useServiceDetails({
     serviceName: fireHydrantServiceName,
+    lookupByName: isFireHydrantAvailable(entity),
```

Comment:
There's a bit of unnecessary complexity here in terms of having to use both getFireHydrantServiceName and isFireHydrantAvailable to additionally convey a boolean - wouldn't it make sense if `useServiceDetails` just accepted an entity as its parameter, and then it does whatever is necessary internally to do its job?

Reply:
I could see that, sure.

I'd prefer to not change `useServiceDetails` at all, but the it currently uses a search instead of a lookup by name/identifier and that complicates things. :/  

It was my attempt to keep the diff easily digestable while preserving existing functionality.